src/main/java/pl/edu/mimuw/logic/FormulaVariable.java

```diff
@@ -0,0 +1,27 @@
+package pl.edu.mimuw.logic;
+
+import java.util.ArrayList;
+
+public class FormulaVariable implements Formula {
```

Comment:
używasz tego jako kluczy w mapie, a nie zaimplementowałeś `equals` i `hashcode` - to nie zadziała zgodnie z oczekiwaniami, tak jak mówiliśmy na ostatnim laboratorium